packages/twenty-front/src/modules/settings/developers/webhook/utils/fetchGraphData.ts

```diff
@@ -0,0 +1,95 @@
+import { NivoLineInput } from '@/settings/developers/webhook/components/SettingsDevelopersWebhookUsageGraph';
+import { WEBHOOK_GRAPH_API_OPTIONS_MAP } from '@/settings/developers/webhook/constants/WebhookGraphAPIOptionsMap';
+import { SnackBarVariant } from '@/ui/feedback/snack-bar-manager/components/SnackBar';
+
+type fetchGraphDataProps = {
+  webhookId: string;
+  enqueueSnackBar: (
+    message: string,
+    options: { variant: SnackBarVariant },
+  ) => void;
+  webhookOptions?: '7D' | '1D' | '12H' | '4H';
+};
+
+export const fetchGraphData = async ({
+  webhookId,
+  enqueueSnackBar,
```

Comment:
It feels a bit strange to pass enqueueSnackBar to a util. Maybe just throw errors here and put your try catch at a high level? You can create a dedicated hook probably?